src/main/java/billgates/view/gui/AddEntryDialog.java

```diff
@@ -0,0 +1,111 @@
+package billgates.view.gui;
+
+import com.github.lgooddatepicker.components.DateTimePicker;
+
+import javax.swing.*;
+import java.awt.*;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+public class AddEntryDialog extends JDialog {
+
+    private int result;
+    private ZonedDateTime date;
+    private static double value;
+    private String currency;
+    private String description;
+    private String from;
+    private String to;
+    private String location;
+    private final MainFrame mainFrame;
+
+    public AddEntryDialog(MainFrame owner) {
+        super(owner);
+        this.mainFrame = owner;
+        // Create a datetime picker.
+        DateTimePicker dateTimePicker1 = new DateTimePicker();
+        add(dateTimePicker1);
+
+        JTextField valueField = new JTextField(20);
+        valueField.setDocument(new RegexDocument("\\S*"));
+        valueField.setDocument(new RegexDocument("\\w*"));
+
+        JTextField currencyField = new JTextField(20);
+        currencyField.setDocument(new RegexDocument("\\S*"));
+        currencyField.setDocument(new RegexDocument("\\w*"));
```

Comment:
I think w* is not really appropriate for the currency field. 

Maybe something like: 
```suggestion
        currencyField.setDocument(new RegexDocument("[a-zA-z]^*"));
```
could work? This RegEx only takes in alphabetical inputs. To be more specific, maybe we can even use `[A-Z]` only, as currency is usually in all caps.

Reply:
For RegEx references, you can check out this link: 
https://docs.oracle.com/javase/tutorial/essential/regex/pre_char_classes.html

Also, I believe in the latest CSC207 lectures, we have covered Regular Expressions, if you need a hand, please call me on WeChat or something :3